backend/app/app/api/api_v1/endpoints/monitoringevent.py

```diff
@@ -1,284 +1,485 @@
+    "/{scsAsId}/subscriptions/{subscriptionId}",
+    response_model=schemas.MonitoringEventSubscription,
+)
 def update_subscription(
     *,
-    scsAsId: str = Path(..., title="The ID of the Netapp that creates a subscription", example="myNetapp"),
+    scsAsId: str = Path(
+        ...,
+        title="The ID of the Netapp that creates a subscription",
+        example="myNetapp",
+    ),
     subscriptionId: str = Path(..., title="Identifier of the subscription resource"),
-    item_in: schemas.MonitoringEventSubscriptionCreate,
+    item_in: schemas.MonitoringEventSubscription,
     current_user: models.User = Depends(deps.get_current_active_user),
-    http_request: Request
+    http_request: Request,
 ) -> Any:
     """
     Update/Replace an existing subscription resource
     """
     db_mongo = client.fastapi
 
     try:
-        retrieved_doc = crud_mongo.read_uuid(db_mongo, db_collection, subscriptionId)
-    except Exception as ex:
-        raise HTTPException(status_code=400, detail='Please enter a valid uuid (24-character hex string)')
-    
-    #Check if the document exists
+        id = ObjectId(subscriptionId)
+    except Exception:
+        raise HTTPException(
+            status_code=400,
+            detail="Please enter a valid uuid (24-character hex string)",
+        )
+
+    filters = {"_id": id}
+    if not user.is_superuser(current_user):
+        filters = {"owner_id": current_user.id}
+    retrieved_doc = db_mongo[db_collection].find_one(
+        filters, {"_id": False, "subscription": True}
+    )
+
+    # Check if the document exists
     if not retrieved_doc:
         raise HTTPException(status_code=404, detail="Subscription not found")
-    #If the document exists then validate the owner
-    if not user.is_superuser(current_user) and (retrieved_doc['owner_id'] != current_user.id):
-        raise HTTPException(status_code=400, detail="Not enough permissions")
 
-    sub_validate_time = tools.check_expiration_time(expire_time=retrieved_doc.get("monitorExpireTime"))
-    
+    sub_validate_time = tools.check_expiration_time(
+        expire_time=retrieved_doc.get("monitorExpireTime")
+    )
+
     if sub_validate_time:
-        #Update the document
-        json_data = jsonable_encoder(item_in)
-        crud_mongo.update_new_field(db_mongo, db_collection, subscriptionId, json_data)
-        
-        #Retrieve the updated document | UpdateResult is not a dict
-        updated_doc = crud_mongo.read_uuid(db_mongo, db_collection, subscriptionId)
-        updated_doc.pop("owner_id")
+        # Update the document
+        retrieved_doc["subscription"] = jsonable_encoder(item_in, exclude_unset=True)
+        crud_mongo.update_new_field(
+            db_mongo, db_collection, subscriptionId, retrieved_doc
+        )
+
+        # Retrieve the updated document | UpdateResult is not a dict
+        updated_doc = crud_mongo.read_uuid(db_mongo, db_collection, subscriptionId)[
+            "subscription"
+        ]
```

Comment:
You can do:
```suggestion
        json_data = jsonable_encoder(item_in, exclude_unset=True)
        updated_doc = db_mongo[db_collection].find_one_and_update(
            {"_id": id},
            {"$set": {"subscription": json_data}},
            projection={"_id": False, "subscription": True},
            return_document=ReturnDocument.AFTER,
        )
```

This will update only the subscription field instead of all fields and return the updated result all in one call.